### PR TITLE
ARROW-11582: [R] write_dataset 'format' argument default and validation could be better

### PR DESCRIPTION
**Pull request by pachadotdev**

### CODE REVIEW

File: r/tests/testthat/test-dataset.R [hunk 1]
```diff
@@ -1061,6 +1061,17 @@ test_that("Writing a dataset: Parquet->Parquet (default)", {
   )
 })
 
+test_that("Writing a dataset: Not specified format", {
+  skip_on_os("windows") # https://issues.apache.org/jira/browse/ARROW-9651
+  dst_dir <- make_temp_dir()
+  write_dataset(mtcars, dst_dir)
+  new_ds <- open_dataset(dst_dir)
```
Comment (jonkeane) [inline r/tests/testthat/test-dataset.R:1068]: Should we assert that this has produced parquet files / the dataset as opened is parquet files?

File: r/tests/testthat/test-dataset.R [hunk 2]
```diff
@@ -1061,6 +1061,17 @@ test_that("Writing a dataset: Parquet->Parquet (default)", {
   )
 })
 
+test_that("Writing a dataset: Not specified format", {
```
Comment (jonkeane) [inline r/tests/testthat/test-dataset.R:1064]:
```suggestion
test_that("Writing a dataset: no format specified", {
```

File: r/tests/testthat/test-dataset.R (hunk 1 above)
Comment (pachamaltese) [reply r/tests/testthat/test-dataset.R:1068]: sure, changing now

Comment (pachamaltese) [reply r/tests/testthat/test-dataset.R:1068]: updated in adf82e4

Comment (nealrichardson) [reply r/tests/testthat/test-dataset.R:1068]: FWIW the default format for open_dataset() is parquet, so this test would have failed if it hadn't written parquet files.

File: r/R/dataset-write.R [hunk 3]
```diff
@@ -56,11 +56,12 @@
 #' @export
 write_dataset <- function(dataset,
                           path,
-                          format = dataset$format,
+                          format = c("parquet", "feather", "arrow", "ipc"),
```
Comment (nealrichardson) [inline r/R/dataset-write.R:59]: Need to update the docs to reflect the new behavior. You may (also) want to borrow the language used to describe `format` from `dataset-format.R`.

**Review by nealrichardson — COMMENTED** [review]
Need to update the docs but otherwise this looks good, thanks!

File: r/R/dataset-write.R (hunk 3 above)
Comment (pachamaltese) [reply r/R/dataset-write.R:58]: how about this?

Comment (nealrichardson) [reply r/R/dataset-write.R:58]: xref to FileFormat is good (if it works, does it need the `()`? FileFormat is not callable) though I think it's important to (still) note that only parquet and feather are supported.

Comment (pachamaltese) [reply r/R/dataset-write.R:58]: fixed! the link it works without ()
image

**Comment by pachamaltese** [comment]
I think it can be merged now, since the default is "parquet"

**Review by nealrichardson — APPROVED** [review]
+1, will merge when CI has passed.

**Pull request closed without merging**